### PR TITLE
chore: run notification locally

### PR DESCRIPTION
**Pull request by pasha-codefresh**
Ability run notification with goreman tool

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -9,3 +9,4 @@ git-server: test/fixture/testrepos/start-git.sh
...
 dev-mounter: [[ "$ARGOCD_E2E_TEST" != "true" ]] && go run hack/dev-mounter/main.go --configmap argocd-ssh-known-hosts-cm=${ARGOCD_SSH_DATA_PATH:-/tmp/argocd-local/ssh} --configmap argocd-tls-certs-cm=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} --configmap argocd-gpg-keys-cm=${ARGOCD_GPG_DATA_PATH:-/tmp/argocd-local/gpg/source}
 applicationset-controller: [ "$BIN_MODE" == 'true' ] && COMMAND=./dist/argocd || COMMAND='go run ./cmd/main.go' && sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_ASK_PASS_SOCK=/tmp/applicationset-ask-pass.sock ARGOCD_SSH_DATA_PATH=${ARGOCD_SSH_DATA_PATH:-/tmp/argocd-local/ssh} ARGOCD_BINARY_NAME=argocd-applicationset-controller $COMMAND --loglevel debug --metrics-addr localhost:12345 --probe-addr localhost:12346 --argocd-repo-server localhost:${ARGOCD_E2E_REPOSERVER_PORT:-8081}"
+notification: [ "$BIN_MODE" == 'true' ] && COMMAND='go run ./cmd/main.go' && sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_BINARY_NAME=argocd-notifications $COMMAND --loglevel debug"
```
Comment (leoluz) [inline Procfile:12]: `BIN_MODE == true` should start the process using the proper binary in the `./dist` folder. It should only use `go run` if `BIN_MODE != true`. This is necessary to be able to properly debug the process as `delve` doesn't work well with processes started with `go run`

**Review by leoluz — CHANGES_REQUESTED** [review]
Please fix the issue with the `COMMAND` variable setup when `BIN_MODE == true`

File: Procfile (hunk 1 above)
Comment (rishabh625) [reply Procfile:12]: Perhaps, why to run  with go run when we have added as command argocd-notifications into Argo-cd binary
We should invoke as argocd-notifications binary name

Also there will be a change in Procfile of test/container/Procfile

**Review by rishabh625 — CHANGES_REQUESTED** [review]

File: test/container/Procfile [hunk 2]
```diff
@@ -11,3 +11,4 @@ nginx: sudo sh -c "test $ARGOCD_E2E_TEST = true && nginx -g 'daemon off;' -c $(p
 helm-registry: sudo sh -c "registry serve /etc/docker/registry/config.yml"
 dev-mounter: test "$ARGOCD_E2E_TEST" != "true" && go run hack/dev-mounter/main.go --configmap argocd-ssh-known-hosts-cm=${ARGOCD_SSH_DATA_PATH:-/tmp/argocd-local/ssh} --configmap argocd-tls-certs-cm=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} --configmap argocd-gpg-keys-cm=${ARGOCD_GPG_DATA_PATH:-/tmp/argocd-local/gpg/source}
 applicationset-controller: sh -c "FORCE_LOG_COLORS=1 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_ASK_PASS_SOCK=/tmp/applicationset-ask-pass.sock ARGOCD_BINARY_NAME=argocd-applicationset-controller go run ./cmd/main.go --loglevel debug  --metrics-addr localhost:12345 --probe-addr localhost:12346 --namespace argocd-e2e --loglevel debug  --argocd-repo-server localhost:${ARGOCD_E2E_REPOSERVER_PORT:-8081}" 
+notification: sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_BINARY_NAME=argocd-notifications go run ./cmd/main.go --loglevel debug"
```
Comment (leoluz) [inline test/container/Procfile:14]: Are there e2e tests added using the notification service?

File: Procfile (hunk 1 above)
Comment (leoluz) [reply Procfile]: tks.. looking good now

File: test/container/Procfile (hunk 2 above)
Comment (pasha-codefresh) [reply test/container/Procfile:14]: No, but i think it worth to add in future, maybe i will take it

**Review by rishabh625 — APPROVED** [review]
Thanks LGTM!!

**Review by crenshaw-dev — APPROVED** [review]
lgtm!

**Review by leoluz — APPROVED** [review]
LGTM

**Pull request merged**